tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -686,22 +678,16 @@ ROOT::Experimental::Detail::RPageSourceDaos::LoadClusters(std::span<RCluster::RK
       std::uint64_t fSize = 0;
    };
 
-   std::vector<unsigned char *> clusterBuffers(clusterKeys.size());
-   std::vector<std::unique_ptr<ROnDiskPageMapHeap>> pageMaps(clusterKeys.size());
-   RDaosContainer::MultiObjectRWOperation_t readRequests;
-
-   int64_t szPayload = 0;
-   unsigned nPages = 0;
-
-   for (unsigned i = 0; i < clusterKeys.size(); ++i) {
-      const auto &clusterKey = clusterKeys[i];
+   /// Prepares read requests for a single cluster; `readRequests` is modified by this function.  Requests are coalesced
+   /// by OID and distribution key.
+   auto fnPrepareSingleCluster = [&](const RCluster::RKey &clusterKey,
```

Comment:
I think I'd prefer this as a named method.

Reply:
I agree, but turning this into a (private) member function would require the definition of `RDaosContainer::MultiObjectRWOperation_t` (the type of the second argument) to be already available in `RPageStorageDaos.hxx`, as C++ does not allow forward-declaring a nested class.

To circumvent that, either _(i)_ we should put an additional `#include <ROOT/RDaos.hxx>`, which itself depends on many DAOS headers, or _(ii)_ a further refactor is required.

Thus, unless you have a strong opinion about this, I'd vote to leave as-is.

Reply:
Ok, then let's add a code comment explaining the reason to use a lambda.